src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -285,21 +285,35 @@ private static unsafe int IndexOfHtmlEncodingChars(string s, int startPos)
 
         #region UrlEncode implementation
         
-        private static void GetEncodedBytes(byte[] originalBytes, int offset, int count, byte[] expandedBytes)
+        private unsafe static void GetEncodedBytes(byte[] originalBytes, int offset, int count, byte[] expandedBytes)
         {
+            Debug.Assert(originalBytes != null);
```

Comment:
> OK; should I do something like throw an ArgumentNullException if it's null in both Debug/Release instead, then?

Yeah, that sounds reasonable.  Previously we would end up throwing NullReferenceException or IndexOutOfRangeException, but changing them to ArgumentNullException and ArgumentOutOfRangeException seems reasonable (given that we never expect these things to be throw and we are adding the checks just to ensure we don't break type safety if things ever go bad).